src/test/unit_tests/test_namespace_fs.js

```diff
@@ -229,6 +234,283 @@ mocha.describe('namespace_fs', function() {
+                }
+            });
+            const obj_nsr = { resource: nsr, path: bucket_path };
+            await rpc_client.bucket.create_bucket({
+                name: bucket_name,
+                namespace: {
+                    read_resources: [obj_nsr],
+                    write_resource: obj_nsr
+                }
+            });
+
+            const disabled_nsr = { resource: nsr, path: disabled_bucket_path };
+            await rpc_client.bucket.create_bucket({
+            name: disabled_bucket_name,
+            namespace: {
+                read_resources: [disabled_nsr],
+                write_resource: disabled_nsr
+            }
+        });
+
+            const policy = {
+                Version: '2012-10-17',
+                Statement: [{
+                    Sid: 'id-1',
+                    Effect: 'Allow',
+                    Principal: { AWS: "*" },
+                    Action: ['s3:*'],
+                    Resource: [`arn:aws:s3:::*`]
+                },
+            ]
+            };
+            // create accounts
+            let res = await nsfs_versioning.generate_nsfs_account({ admin: true });
+            s3_admin = nsfs_versioning.generate_s3_client(res.access_key, res.secret_key);
+            await s3_admin.putBucketPolicy({
+                Bucket: bucket_name,
+                Policy: JSON.stringify(policy)
+            }).promise();
+            await s3_admin.putBucketPolicy({
+                Bucket: disabled_bucket_name,
+                Policy: JSON.stringify(policy)
+            }).promise();
+            // create nsfs account
+            res = await nsfs_versioning.generate_nsfs_account();
+            s3_client = nsfs_versioning.generate_s3_client(res.access_key, res.secret_key);
+            accounts.push(res.email);
+            // create a file in version disabled bucket
+            await create_object(`${disabled_full_path}/${dis_version_key}`, dis_version_body, 'null');
+            // create a file after when versioning not enabled
+            await create_object(`${full_path}/${dis_version_key}`, dis_version_body, 'null');
+            await s3_client.putBucketVersioning({ Bucket: bucket_name, VersioningConfiguration: { MFADelete: 'Disabled', Status: 'Enabled' } }).promise();
+            const bucket_ver = await s3_client.getBucketVersioning({ Bucket: bucket_name }).promise();
+            assert.equal(bucket_ver.Status, 'Enabled');
+            // create base file after versioning enabled
+            file_pointer = await create_object(`${full_path}/${en_version_key}`, en_version_body, '1');
+        });
+
+        mocha.after(async () => {
+            //fs_utils.folder_delete(tmp_fs_root);
+            for (let email of accounts) {
+                await rpc_client.account.delete_account({ email });
+            }
+        });
+
+        mocha.it('get object, versioning not enabled - should return latest object', async function() {
```

Comment:
could you add a test for get/head null version when it's not the latest? 
currently, it's not a different flow than what we have but maybe for future changes it's good to have it as well